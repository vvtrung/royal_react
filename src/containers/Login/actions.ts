import { createActions } from 'redux-actions';

import { ActionTypes } from './constants';

export const { login } = createActions({
  [ActionTypes.LOGIN]: payload => payload,
});
