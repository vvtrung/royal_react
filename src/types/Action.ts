export interface ObjectLiteral {
  [key: string]: any;
}

export type MetaType = {
  setErrors: (errors: ObjectLiteral) => void;
  [key: string]: any;
};

export type ActionPayloadType = {
  values: ObjectLiteral;
  meta: MetaType;
  [key: string]: any;
};

export interface ActionType {
  type: string;
  payload: ActionPayloadType;
}
