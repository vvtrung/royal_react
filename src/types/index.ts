export type { User } from './User';
export type {
  MetaType,
  ActionPayloadType,
  ObjectLiteral,
  ActionType,
} from './Action';
