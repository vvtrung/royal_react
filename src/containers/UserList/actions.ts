import { createActions } from 'redux-actions';

import { User } from 'types';

import { ActionTypes } from './constants';

export const { getUsers, saveUsers } = createActions({
  [ActionTypes.GET_USERS]: payload => payload,
  [ActionTypes.SAVE_USERS]: (users: User) => ({ users }),
});
