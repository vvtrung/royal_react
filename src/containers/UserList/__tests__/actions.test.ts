import { User } from 'types';

import { getUsers, saveUsers } from '../actions';

describe('User Actions', () => {
  it('should create getUsers action', () => {
    expect(getUsers()).toMatchSnapshot();
  });

  it('should create saveUsers action', () => {
    const users: User[] = [
      { id: '1', name: 'name', email: 'email@gmail.com', avatar: 'avatar' },
      { id: '2', name: 'name2', email: 'email2@gmail.com', avatar: 'avatar2' },
      { id: '3', name: 'name3', email: 'email3@gmail.com', avatar: 'avatar3' },
    ];
    expect(saveUsers(users)).toMatchSnapshot();
  });
});
