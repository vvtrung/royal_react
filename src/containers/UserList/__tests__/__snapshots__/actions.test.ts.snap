// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Actions should create getUsers action 1`] = `
Object {
  "type": "GET_USERS",
}
`;

exports[`User Actions should create saveUsers action 1`] = `
Object {
  "payload": Object {
    "users": Array [
      Object {
        "avatar": "avatar",
        "email": "email@gmail.com",
        "id": "1",
        "name": "name",
      },
      Object {
        "avatar": "avatar2",
        "email": "email2@gmail.com",
        "id": "2",
        "name": "name2",
      },
      Object {
        "avatar": "avatar3",
        "email": "email3@gmail.com",
        "id": "3",
        "name": "name3",
      },
    ],
  },
  "type": "SAVE_USERS",
}
`;
